Add note to the guides about action names versus ActionController reserved methods (#42483)
rails/rails: guides/source/action_controller_overview.md

Only public methods are callable as actions. It is a best practice to lower the visibility of methods (with `private` or `protected`) which are not intended to be actions, like auxiliary methods or filters.

> NOTE: While it's possible to define action methods of any name, some method names are reserved by Action Controller. Accidentally redefining them as actions, or even as auxiliary methods, could result in `SystemStackError` or other exceptions from any route the controller services. Controllers that limit their actions to only the default RESTful [Resource Routing][] actions do not need to worry about this, but if you define custom routes on your controller you should be aware of the potential for collision with a reserved method. If you find that your application would benefit from using a reserved method as an action name, one workaround is to define your action with a non-reserved method name and then use a custom route to map the otherwise reserved method name to your non-reserved action method.

Reviewer: ```suggestion
WARNING: Some method names are reserved by Action Controller. Accidentally redefining them as actions, or even as auxiliary methods, could result in `SystemStackError`. Limiting your controllers to only RESTful [Resource Routing][] actions should not need to worry about this.

NOTE: If you must use a reserved method as an action name, one workaround is to use a custom route to map the reserved method name to your non-reserved action method.
```

Reviewer: @chrisbloom7 Some wording changes if you could make those and squash your commits I will merge :bow:
